dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESClient.java

```diff
@@ -209,159 +195,125 @@ public void setReplicasSettings () {
+        externalSettings.put(ServerPort.ES_TRANSPORT_TCP_PORT.getPropertyName(), transportTCPPort);
 
-			}
+        if(UtilMethods.isSet(externalSettings.get("http.enabled")) && (Boolean.parseBoolean(externalSettings.get("http.enabled")))) {
+            httpPort = UtilMethods.isSet(currentServer.getEsHttpPort()) ? currentServer.getEsHttpPort().toString()
+                    :ClusterFactory.getNextAvailablePort(serverId, ServerPort.ES_HTTP_PORT, externalSettings);
+            currentServer = Server.builder(currentServer).withEsHttpPort(Integer.parseInt(httpPort)).build();
+            if (!UtilMethods.isSet(externalSettings.get("http.host"))){
+                externalSettings.put("http.host", bindAddr);
+            }
 
-			List<String> myself = new ArrayList<String>();
-			myself.add(currentServer.getServerId());
+            externalSettings.put(ServerPort.ES_HTTP_PORT.getPropertyName(), httpPort);
+        }
+        List<String> myself = new ArrayList<>();
+        myself.add(currentServer.getServerId());
 
-			List<Server> aliveServers = serverAPI.getAliveServers(myself);
+        currentServer = Server.builder(currentServer).withEsTransportTcpPort(Integer.parseInt(transportTCPPort)).build();
 
-			currentServer = Server.builder(currentServer).withEsTransportTcpPort(Integer.parseInt(transportTCPPort)).build();
-			aliveServers.add(currentServer);
+        setUnicastHosts(myself, transportTCPPort, bindAddr, serverAPI, currentServer,
+                externalSettings);
 
-			StringBuilder initialHosts = new StringBuilder();
+        try {
+            serverAPI.updateServer(currentServer);
+        } catch (DotDataException e) {
+            Logger.error(this, "Error trying to update server. Server Id: " + currentServer.getServerId());
+        }
 
-			int i=0;
-			for (Server server : aliveServers) {
-				if(i>0) {
-					initialHosts.append(",");
-				}
+        shutDownNode();
+		initNode(externalSettings);
+	}
 
-				if(UtilMethods.isSet(server.getHost()) && !server.getHost().equals("localhost")) {
-					initialHosts.append(server.getHost()).append(":").append(server.getEsTransportTcpPort());
-				} else {
-					initialHosts.append(server.getIpAddress()).append(":").append(server.getEsTransportTcpPort());
-				}
+    private void setUnicastHosts(final List<String> myself,
+            final String transportTCPPort, final String bindAddr, final ServerAPI serverAPI,
+            final Server currentServer, final Builder externalSettings) throws DotDataException {
 
-				i++;
-			}
+        String initData;
 
-			if(initialHosts.length()==0) {
-				if(bindAddr.equals("localhost")) {
-					initialHosts.append(currentServer.getIpAddress()).append(":").append(transportTCPPort);
-				} else {
-					initialHosts.append(bindAddr).append(":").append(transportTCPPort);
-				}
-			}
+        List<Server> aliveServers = serverAPI.getAliveServers(myself);
+        aliveServers.add(currentServer);
```

Comment:
if we call the `serverAPI.updateServer(currentServer);` before this method `setUnicastHosts` , could we just call `serverAPI.getAliveServers()` (no exclusion) instead of excluding myself and then adding me?